telemetry/src/lib.rs

```diff
@@ -46,20 +81,112 @@ impl TelemetryConfig {
         }
     }
 
-    pub fn try_init(self) -> Result<(), tracing_subscriber::util::TryInitError> {
+    /// Enable OpenTelemetry/OTLP tracing.
+    /// You can optionally provide a service name, version, and environment.
+    /// If omitted, these will default to environment variables (`SERVICE_NAME`, `SERVICE_VERSION`, `ENVIRONMENT`) or hard-coded strings.
+    #[must_use]
+    pub fn with_opentelemetry(
+        mut self,
+        service_name: Option<String>,
+        service_version: Option<String>,
+        environment: Option<String>,
+    ) -> Self {
+        self.use_otel = true;
+        self.service_name = service_name;
+        self.service_version = service_version;
+        self.environment = environment;
+        self
+    }
+
+    /// Initialize the OpenTelemetry TracerProvider and set it globally, returning it for storing.
+    fn init_opentelemetry(&mut self) -> Result<opentelemetry_sdk::trace::TracerProvider, Box<dyn std::error::Error>> {
+        // Fallback to environment variables if the user did not supply them.
+        let default_service_name =
```

Comment:
all configuration for a library should be passed into the library explicitly, not through environment variables. End applications should have control over which environment variables they read and its more clear to have that in the main function of the users of this code.